rnd/autogpt_builder/src/components/monitor/FlowRunInfo.tsx

```diff
@@ -34,12 +40,19 @@ export const FlowRunInfo: React.FC<
             Run ID: <code>{flowRun.id}</code>
           </p>
         </div>
-        <Link
-          className={buttonVariants({ variant: "outline" })}
-          href={`/build?flowID=${flow.id}`}
-        >
-          <Pencil2Icon className="mr-2" /> Edit Agent
-        </Link>
+        <div className="flex space-x-2">
+          {flowRun.status === "running" && (
+            <Button onClick={handleStopRun} variant="destructive">
```

Comment:
You can wrap this Button in <form action={handleStopRun}> to make this a server action that doesn’t run api calls from the front end which would be appreciated

Reply:
If you need to pass data to it you can bind it

Reply:
wrap in what?

Reply:
suggested below the actual changes.

See: 
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind
- https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#examples
- https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#useeffect

Reply:
also lmao github removed the text of the HTML tag when rendering above